libraries/AP_DroneCAN/AP_DroneCAN.cpp

```diff
@@ -1503,6 +1503,58 @@ bool AP_DroneCAN::is_esc_data_index_valid(const uint8_t index) {
     return true;
 }
 
+#if AP_SCRIPTING_ENABLED
+/*
+  handle FlexDebug message, holding a copy locally for a lua script to access
+ */
+void AP_DroneCAN::handle_FlexDebug(const CanardRxTransfer& transfer, const dronecan_protocol_FlexDebug &msg)
+{
+    // find an existing element in the list
+    const uint8_t source_node = transfer.source_node_id;
+    for (auto *p = flexDebug_list; p != nullptr; p = p->next) {
+        if (p->node_id == source_node && p->msg.id == msg.id) {
+            p->msg = msg;
+            p->timestamp_us = uint32_t(transfer.timestamp_usec);
+            return;
+        }
+    }
+
+    // new message ID, add to the list. Note that this gets called
+    // only from one thread, so no lock needed
+    auto *p = NEW_NOTHROW FlexDebug;
```

Comment:
Concerned that DroneCAN devices can cause arbitrary amounts of flight controller memory to be consumed that is never released. I think there should be a Lua call that allocates a fixed size buffer (or at least sets a limit on the number of allowed allocations) like with the scripting CAN buffers.

Reply:
I don't think it will be a problem in practice, as amount of memory for a scripting capable flight controller isn't much
we could reduce the ram by only allocating up to the actual length of the message

Reply:
Fixed by adding the option bit so it's not possible for this to cause problems with ordinary users.